contracts/amm/stableswap/factory/factory_v_100.vy

```diff
@@ -110,19 +112,31 @@ fee_receiver: public(address)
 markets: HashMap[uint256, address[4294967296]]
 market_counts: HashMap[uint256, uint256]
 
+deployer: immutable(address)
 
 @external
 def __init__(_fee_receiver: address, _owner: address):
 
     self.fee_receiver = _fee_receiver
-    self.admin = _owner
+    self.admin = msg.sender
+    deployer = msg.sender
 
     self.asset_types[0] = "Standard"
     self.asset_types[1] = "Oracle"
     self.asset_types[2] = "Rebasing"
     self.asset_types[3] = "ERC4626"
 
 
+@external
+def set_owner(_owner: address):
+    
+    assert msg.sender == deployer
+    assert self.admin == deployer
+    assert _owner != deployer
+
+    self.admin = _owner
```

Comment:
Maybe we want an ability to change owners more than 1 time? For example, we might update DAO contract

Reply:
you can do it via: commit transfer ownership > accept transfer ownership. i think deployer should only get one chance for setting everything up.